docs/guides/upstream-mtls.md

```diff
@@ -0,0 +1,158 @@
+
+1. In a terminal environment and the same directory where you created the certificate files, start an OpenSSL server process. Note that it will run in the foreground until stopped, so you will need another terminal environment to run additional commands:
+
+    ```bash
+    openssl s_server -key ./openssl.localhost-key.pem -cert ./openssl.localhost.pem -accept 44330 -www
+    ```
+
+    You can confirm that the server is responding using `curl`:
+
+    ```
+    curl -k https://localhost:44330
+    <HTML><BODY BGCOLOR="#ffffff">
+    <pre>
+
+    s_server -key ./openssl.localhost-key.pem -cert ./openssl.localhost.pem -accept 44330 -www
+    Secure Renegotiation IS supported
+    Ciphers supported in s_server binary
+    ...
+    ```
+
+    ::: tip
+    Even though OpenSSL is providing a certificate signed by a CA in our trust store, we need the `-k` flag because the domain name `localhost` doesn't match `openssl.localhost` defined in our certificate.
+    :::
+
+## Configure Pomerium
+
+1. Update the `config.yaml` file in `/etc/pomerium` with a new route for this upstream service:
+
+    ```yaml
+    - from: https://openssl.pomerium.localhost.io
+      to: https://localhost:44330
+      tls_server_name: openssl.localhost
+      policy:
+        - allow:
+            or:
+              - email:
+                  is: user@example.com
+    ```
+
+    Pomerium should automatically process changes when the configuration file is updated.
+
+1. Visit <https://openssl.localhost.pomium.io> in your browser to confirm the route:
+
+    ![OpenSSL Upstream without client certificate verification](./img/mtls/openssl-test1.png)
+
+    Note the last line:
+
+    > no client certificate available
+
+    In the next section we'll adjust the OpenSSL server to require a client certificate and edit our Pomerium configuration to provide one.
+
+## Install Pomerium Client Certificate
+
+1. Stop the OpenSSL server process (**Ctrl+C**) and start a new one with the additional flag `-Verify 1`:
+
+    ```bash
+    openssl s_server -Verify 1 -key ./openssl.localhost-key.pem -cert ./openssl.localhost.pem -accept 44330 -www
+    ```
+
+1. When you refresh <https://openssl.localhost.pomium.io> in your browser, the connection will fail. Back in the terminal, the OpenSSL server should output errors containing:
+
+    ```log
+    SSL routines:tls_process_client_certificate:peer did not return a certificate
+    ```
+
+    This confirms that the upstream service is now requiring a trusted client certificate to allow connections.
+
+1. Update the Pomerium route block to provide the client certificate:
+
+    ```yaml
+    - from: https://openssl.pomerium.localhost.io
+      to: https://localhost:44330
+      tls_server_name: openssl.localhost
+      tls_client_cert_file: /etc/pomerium/pomerium@localhost-client.pem
+      tls_client_key_file: /etc/pomerium/pomerium@localhost-client-key.pem
```

Comment:
```suggestion
      tls_client_cert_file: pomerium@localhost-client.pem
      tls_client_key_file: pomerium@localhost-client-key.pem
```

Reply:
Per discussion, this is required by the assumed environment, as with the other outstanding comment.